sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -4129,6 +4129,36 @@ class Dataset[T] private[sql](
     new DataFrameWriterV2[T](table, this)
   }
 
+  /**
+   * Create a [[MergeIntoWriter]] for MergeInto action.
```

Comment:
This is user facing API doc. Not sure if it is proper to put `MergeIntoWriter` there. For example, we don't put `DataFrameWriter` in `write` API doc.

Reply:
Maybe just describing what the function is used for. E.g., "Merges a set of updates, insertions, and deletions based on a source table into a target table"

https://docs.databricks.com/en/sql/language-manual/delta-merge-into.html

Reply:
The key is the `MergeIntoWriter` is public API or developer API.
cc @cloud-fan

Reply:
Fixed. Thanks